drivers/gles2/rasterizer_scene_gles2.cpp

```diff
@@ -2078,6 +2078,12 @@ void RasterizerSceneGLES2::_setup_light(LightInstance *light, ShadowAtlas *shado
 				//	state.scene_shader.set_uniform(SceneShaderGLES2::LIGHT_CLAMP, light_clamp);
 				state.scene_shader.set_uniform(SceneShaderGLES2::SHADOW_PIXEL_SIZE, Size2(1.0 / directional_shadow.size, 1.0 / directional_shadow.size));
 				state.scene_shader.set_uniform(SceneShaderGLES2::LIGHT_SPLIT_OFFSETS, split_offsets);
+
+				const float fade_start = light_ptr->param[VS::LIGHT_PARAM_SHADOW_FADE_START];
+				// Using 1.0 would break `smoothstep()` in the shader.
+				state.scene_shader.set_uniform(SceneShaderGLES2::FADE_FROM, -split_offsets[shadow_count - 1] * MIN(fade_start, 0.999));
+				state.scene_shader.set_uniform(SceneShaderGLES2::FADE_TO, -split_offsets[shadow_count - 1]);
+
```

Comment:
Do we need to send this? Can we use one of the `light_split_offsets` in the shader for the furthest value?

Reply:
I've copied the 4.x approach, I'm not sure if we can reduce the number of uniforms sent without changing how it looks.

Reply:
Well for `SceneShaderGLES2::FADE_TO` I couldn't help noticing we have already sent this value 5 lines above (as `split_offsets`), and each uniform is additional extra API (especially in GLES2), and can compete for limited space in GPU for highly cached stuff.

The only question is which of these 4 is the furthest split, but I'm presuming this must have been previously calculated in the shader (for it to have worked previously).

There are 2 questions that appear:
* Is it more expensive to use the existing `if` clauses than to send as an independent additional uniform (this may be the case, dynamic branches aren't good but maybe we have paid the price already)
* The existing `light_split_offset` is stored as `highp`. This might not be necessary for the calculation so could maybe be downcast.

I guess these can only be answered by performance testing especially on mobile. And I guess could be left to a follow up. I'm not sure whether this was considered in 4.x, but maybe there were additional factors at play.